Overhaul Site Package tutorial (#143)
TYPO3-Documentation/TYPO3CMS-Tutorial-SitePackage: Documentation/ExtensionConfiguration/Index.rst

.. include:: /CodeSnippets/ExtensionConfiguration/ComposerJson.rst.txt

For historic reasons TYPO3 extension names are written in

> lower case words and separated by underscores if there are more than one. This

Reviewer: "if there are more than one" - why this addition?

Author: Hi Chris I wanted to stress that extension names can be 1 word or they can be more than 1 word. Maybe I wasn't very clear

> lower case words and separated by underscores if there are more than one. This

Author: ```suggestion
lower case words and separated by underscores. This
```

> is known as the extension key.The directory containing the extension should have

Reviewer: ```suggestion
is known as the extension key. The directory containing the extension should have
```

> is known as the extension key.The directory containing the extension should have

Reviewer: ```suggestion
is known as the extension key. The directory containing the extension should have
```

the same name as the extension key. Composer package names are written in
lower-case words but are separated with dashes if there is more than one word.

> At the top of the composer.json file we see the composer package name

Reviewer: ```suggestion
At the top of the composer.json file we see the Composer package name
```

> At the top of the composer.json file we see the composer package name

Reviewer: ```suggestion
At the top of the :file:`composer.json` file we see the composer package name
```

> At the top of the :file:`composer.json` file we see the composer package name

Reviewer: ```suggestion
At the top of the :file:`composer.json` file we see the Composer package name
```

> :file:`t3docs/site-package` (with a dash) and at the bottom we see the TYPO3

Reviewer: ```suggestion
`t3docs/site-package` (with a dash) and at the bottom we see the TYPO3
```

extension key in the extras section - :file:`site_package` (with an underscore).
The Composer "name" consists of a vendor name followed by a forward slash and the

> lowercase extension name with dashes.

Reviewer: It does not have to be the extension key with dashes. Actually, it can be any name and may have nothing in common with the extension key. I am doing this in my extensions, for example, "matomo_widgets" as extension key, and "typo3-matomo-widgets" as part of the Composer package name.

If we let this as is, please adjust "extension name" to "extension key"

Author: Hi Chris I found this really confusing in the documentation when I tried to follow it. I only wanted it to work and so the simplest thing for me was just to have site-package and site_package. Anything else was just confusing at this point (for me)

.. hint::

>    Make sure you don't mix up your underscores and dashes otherwise composer

Reviewer: ```suggestion
   Make sure you do not mix up your underscores and dashes otherwise Composer
```

>    Make sure you don't mix up your underscores and dashes otherwise composer

Reviewer: ```suggestion
   Make sure you don't mix up your underscores and dashes otherwise Composer
```

   will not find your site-package extension.

.. _extension-manager-configuration:

Extension declaration file :file:`ext_emconf.php`
=================================================

From version 11 this file is only mandatory if you are not using Composer.
If you are using Composer you can omit this file as it is ignored by TYPO3
anyway.

> If you are using a TYPO3 version below 11 or can't use

Reviewer: Additionally: Also when planned to publish it on TER, an ext_emconf.php file is necessary,

    site_package/Resources/Public/JavaScript
    site_package/Resources/Public/JavaScript/website.js

> At this point we can install the sitepackage extension in an TYPO3 instance,

Reviewer: Really "an TYPO3"?
